astroid/brain/brain_typing.py

```diff
@@ -140,6 +157,150 @@ def infer_typing_typevar_or_newtype(node, context_itton=None):
     return node.infer(context=context_itton)
 
 
+def infer_typing_newtype(
+    node: nodes.Call, context_itton: typing.Optional[context.InferenceContext] = None
+) -> typing.Iterator[nodes.ClassDef]:
+    """Infer a typing.NewType(...) call"""
+    try:
+        func = next(node.func.infer(context=context_itton))
```

Comment:
If there is any ambiguity in the inference then we fail to notice that here. Perhaps we should use `helpers.safe_infer`?

Reply:
Ran into a small issue with this. Since `decimal.Decimal` has two implementations (`_pydecimal` vs `_decimal`) it always seems to end up ambiguous.

Is this the desired behavior for cases like this? If so I'll make the change and rework the tests, since I chose `Decimal` as a test type arbitrarily anyway, but I figured I'd confirm.